S2VX.Game.Tests/CommandTests.cs

```diff
@@ -0,0 +1,150 @@
+using osu.Framework.Utils;
+using osuTK;
+using osuTK.Graphics;
+using S2VX.Game.Story.Command;
+using System;
+
+namespace S2VX.Game.Tests {
+    [TestFixture]
+    public class CommandTests {
+        [Test]
+        public void FromString_DoubleCommand() {
+            var input = "ApproachesDistance|0.1|0.2|None|0.1|1.0";
+            var expected = new ApproachesDistanceCommand() {
+                StartTime = 0.1f,
+                EndTime = 0.2f,
+                Easing = Enum.Parse<Easing>("None"),
+                StartValue = 0.1f,
+                EndValue = 1.0f,
+            };
+            var actual = (ApproachesDistanceCommand)S2VXCommand.FromString(input);
+            var tolerance = 0.001;
+            Assert.AreEqual(expected.StartTime, actual.StartTime, tolerance);
+            Assert.AreEqual(expected.EndTime, actual.EndTime, tolerance);
+            Assert.AreEqual(expected.Easing, actual.Easing);
+            Assert.AreEqual(expected.StartValue, actual.StartValue, tolerance);
+            Assert.AreEqual(expected.EndValue, actual.EndValue, tolerance);
+        }
+
+        [Test]
+        public void FromString_ColorCommand() {
+            var input = "BackgroundColor|0.1|0.2|None|(0,0,0)|(1,1,1)";
+            var expected = new BackgroundColorCommand() {
+                StartTime = 0.1f,
+                EndTime = 0.2f,
+                Easing = Enum.Parse<Easing>("None"),
+                StartValue = Color4.Black,
+                EndValue = Color4.White,
+            };
+            var actual = (BackgroundColorCommand)S2VXCommand.FromString(input);
+            var tolerance = 0.001;
+            Assert.AreEqual(expected.StartTime, actual.StartTime, tolerance);
+            Assert.AreEqual(expected.EndTime, actual.EndTime, tolerance);
+            Assert.AreEqual(expected.Easing, actual.Easing);
+            Assert.AreEqual(expected.StartValue, actual.StartValue);
+            Assert.AreEqual(expected.EndValue, actual.EndValue);
+        }
+
+        [Test]
+        public void FromString_Vector2Command() {
+            var input = "CameraMove|0.1|0.2|None|(0,0)|(1,1)";
+            var expected = new CameraMoveCommand() {
+                StartTime = 0.1f,
+                EndTime = 0.2f,
+                Easing = Enum.Parse<Easing>("None"),
+                StartValue = Vector2.Zero,
+                EndValue = Vector2.One,
+            };
+            var actual = (CameraMoveCommand)S2VXCommand.FromString(input);
+            var tolerance = 0.001;
+            Assert.AreEqual(expected.StartTime, actual.StartTime, tolerance);
+            Assert.AreEqual(expected.EndTime, actual.EndTime, tolerance);
+            Assert.AreEqual(expected.Easing, actual.Easing);
+            Assert.AreEqual(expected.StartValue, actual.StartValue);
+            Assert.AreEqual(expected.EndValue, actual.EndValue);
+        }
+
+        [Test]
+        public void ValueAt_Double_0StartTime0DurationUsesStartValue() {
```

Comment:
I think these ValueAt tests are better off in a separate InterpolationTests.cs file, since it looks like it's the Interpolation that is being tested, not Commands.

Reply:
Now that I think about it, since it's an osu! framework command, we shouldn't need to have unit tests for them at all. Should I just remove them?

Reply:
As discussed, looks like osu! framework doesn't have a lot of unit tests for these edge cases, so we'll keep our tests for ourselves